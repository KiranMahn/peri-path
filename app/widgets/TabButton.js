import { TouchableOpacity, Text, StyleSheet } from "react-native";

const TabButton = ({text, isSelected}) => {
    return (
        <TouchableOpacity
        style={{
            borderRadius: 0,
            backgroundColor: '#f8f8f9',
            color: (isSelected ? '#009688' : 'grey'),
            fontWeight: (isSelected ? 'bolder' : 'unset'),
            borderBottomStyle: 'solid',
            borderBottomColor: (isSelected ? '#009688' : '#d8d8d8'),
            borderBottomWidth: (isSelected ? 'thick' : 'thin'),
            margin: 0,
            padding: '0.5em',
        }}
        >
            <Text>{text}</Text>
        </TouchableOpacity>
    );
}

export default TabButton;