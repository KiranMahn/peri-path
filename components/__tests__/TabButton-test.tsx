import React from 'react';
import renderer from 'react-test-renderer';
import TabButton from '../../app/widgets/TabButton';

test('renders correctly', () => {
  const tree = renderer.create(<TabButton text={undefined} isSelected={undefined} />).toJSON();
  expect(tree).toMatchSnapshot();
});