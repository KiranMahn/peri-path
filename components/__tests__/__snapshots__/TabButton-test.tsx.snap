// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#f8f8f9",
      "borderBottomColor": "#d8d8d8",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "thin",
      "borderRadius": 0,
      "color": "grey",
      "fontWeight": "unset",
      "margin": 0,
      "opacity": 1,
      "padding": "0.5em",
    }
  }
>
  <Text />
</View>
`;
